rtb/analisi_dei_requisiti_ver0.4.0.typ

```diff
@@ -0,0 +1,732 @@
+- *Estensione*: Errore "Credenziali errate" (#link(<uc8>)[UC8]).
+
+=== UC8 - Errore "Credenziali errate" <uc8>
+- *Attori principali*: Utente non autenticato.
+- *Precondizioni*: l'utente non autenticato tenta di accedere alla _dashboard_ con delle credenziali errate.
+- *Postcondizioni*: l'utente non autenticato riceve un messaggio di errore.
+- *Trigger*: l'utente non autenticato inserisce delle credenziali errate.
+- *Scenario principale*:
+  + L'utente inserisce delle credenziali errate durante la fase di autenticazione alla _dashboard_.
+  + L'utente riceve un messaggio di errore, segnalando il fatto che le credenziali di accesso inserite siano invalide e invitando a riprovare l'autenticazione.
+#v(20pt)
+#figure(
+  image("../assets/use_cases/UC7-8.svg"),
+  caption: [Diagramma dei casi d'uso UC7 e UC8],
+)
+
+#pagebreak()
+
+=== UC9 - Visualizzazione della mappa dei mezzi noleggiati
+- *Attore principale*: Amministratore.
+- *Precondizioni*: l'amministratore del sistema è autenticato e ha accesso alla _dashboard_ del sistema.
+- *Postcondizioni*: l'amministratore ottiene una visione della posizione e del movimento dei mezzi attualmente in uso all'interno di una mappa.
+- *Trigger*: l'amministratore intende visualizzare la posizione dei mezzi noleggiati.
+- *Scenario principale*:
+  + L'amministratore è collegato e autenticato alla _dashboard_ del sistema.
+  + La _dashboard_ mette a disposizione una mappa interattiva con i mezzi attualmente a noleggio, la cui posizione viene indicata attraverso dei _#rifGlossario("marker")_.
+#v(20pt)
+#figure(
+  image("../assets/use_cases/UC9.svg"),
+  caption: [Diagramma del caso d'uso UC9],
+)
+
+#pagebreak()
+
+=== UC10 - Visualizzazione degli annunci pubblicitari generati lato amministratore
+- *Attore principale*: Amministratore.
+- *Precondizioni*: l'amministratore è autenticato al sistema.
+- *Postcondizioni*: l'amministratore visualizza gli annunci generati dalla LLM per ogni utente con un noleggio attivo all'interno della mappa.
+- *Trigger*: l'amministratore vuole prendere visione di quanto generato da parte della LLM per i clienti del noleggio e per i punti di interesse convenzionati.
+- *Scenario principale*:
+  + L'amministratore accede alla mappa dei mezzi noleggiati.
+  + L'amministratore, tramite un "_click_" sul _marker_ che indica la posizione di un mezzo, visualizza l'annuncio pubblicitario generato per l'utente che ha attualmente in uso il mezzo.
+#v(20pt)
+#figure(
+  image("../assets/use_cases/UC10.svg"),
+  caption: [Diagramma del caso d'uso UC10],
+)
+
+//TODO: valutare se raffinare i casi d'uso con dei sottocasi
+
+//UC Perruz
+
+/*
+L'attore sarebbe il sistema => no UC
+Domanda: il capitolato chiede di avvisare l'admin, si può immaginare tramite notifica, però l'attore della notifica è il sistema quindi non andrebbe creato il caso d'uso. Come si fa ad implementare?
```

Comment:
In caso di "visualizzazione" l'attore principale è chi visualizza (es. tutti i casi d'uso di "visualizza errore")

Reply:
Allora. Ho una conferma abbastanza da Cardin per quanto riguarda questo: ovviamente tu con i diagrammi UC non puoi modellare il funzionamento di sistema (di fatto prossima settimana bisogna fixare degli Use Case che abbiamo tirato su io e davide in quanto non hanno senso di esistere, quindi attenderei anche un attimo a fare i grafici secondo me @ErPreco @Perruzz altrimenti diventa na bestemmia modificare tutto... aggiungo che mi è stato detto che è possibile farli con Typst, quindi prossima settimana mi sa che lavorerò a ciò). Quindi: come avvisare a noi non ce ne può fregare di meno negli UC. L'unica cosa che va fatta è la possibilità di "Visualizzare la notifica"